crates/runtime/src/lib.rs

```diff
@@ -53,20 +93,90 @@ impl Runtime {
     pub fn new(
         config: Config,
         app: app::App,
+        auth: auth::AuthProviders,
         df: DataFusion,
         models: HashMap<String, Model>,
         pods_watcher: podswatcher::PodsWatcher,
     ) -> Self {
         Runtime {
             app: Arc::new(app),
             config,
-            df: Arc::new(df),
+            auth: Arc::new(auth),
+            df: Arc::new(RwLock::new(df)),
             models: Arc::new(models),
             pods_watcher,
         }
     }
 
-    pub async fn start_servers(&self) -> Result<()> {
+    pub async fn load_dataset(
```

Comment:
Should this be a mutable self function? Then we don't need to pass the auth providers

Reply:
load_dataset is used [here](https://github.com/spiceai/spiceai/pull/689/files#diff-c1cad500b94aa50ce0f9f728ca73e6c4be703c15e5e5cc64f5b5644490747fe1R77) before runtime is created. I decided to keep this behavior as-is to minimize the changes and potential impact => thus static function instead of instance method. We can update this to create runtime first and them load datasets